uaclient/cli.py

```diff
@@ -1853,20 +1853,24 @@ def setup_logging(console_level, log_level, log_file=None, logger=None):
     console_handler.set_name("ua-console")  # Used to disable console logging
     logger.addHandler(console_handler)
 
-    # Setup file logging
-    if util.we_are_currently_root():
-        # Setup readable-by-root-only debug file logging if running as root
-        log_file_path = pathlib.Path(log_file)
+    log_file_path = pathlib.Path(log_file)
 
-        if not log_file_path.exists():
-            log_file_path.touch()
+    if not log_file_path.exists():
+        log_file_path.parent.mkdir(parents=True, exist_ok=True)
+        log_file_path.touch()
+        if util.we_are_currently_root():
             log_file_path.chmod(0o644)
+    # Setup file logging
 
+    try:
         file_handler = logging.FileHandler(log_file)
         file_handler.setFormatter(JsonArrayFormatter())
         file_handler.setLevel(log_level)
         file_handler.set_name("ua-file")
         logger.addHandler(file_handler)
+    except PermissionError as e:
+        logging.error(e)
+        return 1
```

Comment:
`setup_logging`  doesn't have a return type and we don't use a return value where it's called AFAIK

When do we expect this exception to occur? a `logging.warning` might be sufficient

Reply:
Actually we were able to solve the problem , when the user log file is in /var/ folder, so no reason for the try...except clause.